models/issue.go

```diff
@@ -1834,7 +1834,7 @@ type DependencyInfo struct {
 // Get Blocked By Dependencies, aka all issues this issue is blocked by.
 func (issue *Issue) getBlockedByDependencies(e Engine) (issueDeps []*DependencyInfo, err error) {
 	err = e.Table("issue_dependency").
-		Select("issue.id, issue.repo_id, issue.index, issue.is_closed, issue.name").
+		Select("issue.*").
```

Comment:
Just asking... is it necessary to select any columns from `Issue` to build a `DependencyInfo`?

Reply:
Thanks @guillep2k for taking a look at this. I "think" it is necessary to pull info from the issues table to properly get the relevant dependency info (i.e. that is needed for display purposes). Originally I was only selecting the necessary columns, but I changed to issue.* because it was breaking the build in mysql and sqlite (though I'm not sure why, it was working locally with mysql). I'm totally open for suggestions thought. I'm new to this project and go so I could definitely be missing the optimal solution. Thanks again!

Reply:
I think it's like a shortcut you're taking, there. You're forcing xorm to bring all the related issues for displaying, but perhaps it's not the most expected place to do that. The fact that you've been asking for some data is what might have confused the ORM. I'm no expert on it either. Have you tried selecting no columns at all from `Issues`? I can't check ATM.

Reply:
I tried selecting no columns and it worked. I also tried just leaving the original code which was 'Select("issue.*")', so I'll just leave that unless you feel I should remove it, thanks

Reply:
Oh, I see, you did not introduce that line. It seems to be just unnecesary code; perhaps an attempt to bring the rows to the ORM ahead of time, but I made some tests with and without that line and it doesn't change the number of queries made to the database. So, I feel that this is what we should be doing:
```
	err = e.Table("issue_dependency").
		Join("INNER", "issue", "issue.id = issue_dependency.dependency_id").
		Where("issue_id = ?", issue.ID).
		Find(&issueDeps)
```

Reply:
thanks @guillep2k I've updated it

Reply:
well haha actually that broke it, the id in my struct is being filling with the id from the issue_dependency table instead of the id from the issue table that I need. From taking a look at the xorm join docs, the returned columns need to be in the right order, so I probably need to re-structure the queries, or just use a straight sql in xorm, but straight sql seems to be used very sparingly in gitea so I should probably do the former.

Reply:
If ever possible, direct SQL should be the last resort. :grin:

Reply:
I changed it to a more xorm friendly query I believe. for example: 
`	err = e.Table("issue").
		Join("INNER", "repository", "repository.id = issue.repo_id").
		Join("INNER", "issue_dependency", "issue_dependency.dependency_id = issue.id").
		Where("issue_id = ?", issue.ID).
		Find(&issueDeps)
`